### DISCUSSION TITLE
v1.5.4

### DISCUSSION THREAD

**Discussion by The-Best-Codes** [The-Best-Codes/codequill#15] [Announcements]
![CodeQuill Social Cover](https://github.com/user-attachments/assets/aa659a0a-379d-43f7-ac0a-0f12ec3539b8)

----

## What's Changed
* New Crowdin updates by @The-Best-Codes in https://github.com/The-Best-Codes/codequill/pull/12
* New Crowdin updates by @The-Best-Codes in https://github.com/The-Best-Codes/codequill/pull/13
* New Crowdin updates by @The-Best-Codes in https://github.com/The-Best-Codes/codequill/pull/14

## New Features:

In this release, I'm excited to announce several new features! Here's a few of them:

### Settings
I've now added a settings pane with a few settings options, including UI language and default editor language.

![settings](https://github.com/user-attachments/assets/6c7143ce-54bd-4842-a195-78a807e22ef5)

### AI name generation
AI name generation is now 100% free and unlimited with no account required. To use it, click the wand icon on your project.

![ai_name_wand](https://github.com/user-attachments/assets/8dd71764-12bf-41e2-86a1-a7782477f20d)

### New Programming Language Support
CodeQuill support more than 30 new coding languages that it did not support before. I'm planning to implement a search dropdown soon to make it a bit easier to find them...

![lang_options](https://github.com/user-attachments/assets/505bba66-ede1-4452-97f1-b61b9eb119a0)

### Minor UI Improvements
I adjusted the UI and scroll areas a bit, made some buttons more responsive, and made the search feature more intuitive.

### :tada: JavaScript Console!
You can now easily run and debug your JS codes, just like you can live preview your HTML ones. Check out the video below to see how it works:

https://github.com/user-attachments/assets/14a3f2a9-2ffb-4f0d-9d5b-f1688888de35

----

## Thank you for using CodeQuill!

**Full Changelog**: https://github.com/The-Best-Codes/codequill/compare/v1.3.2...v1.5.4

<hr /><em>This discussion was created from the release <a href='https://github.com/The-Best-Codes/codequill/releases/tag/v1.5.4'>v1.5.4</a>.</em>

**Comment by Alternate-Existance** [comment]
:fire: Good as usual. Keep it up!

**Reply by The-Best-Codes** [reply]
Thanks for your support! Always be sure to let me know if you have any feature ideas.
